vm/src/stdlib/scproxy.rs

```diff
@@ -0,0 +1,140 @@
+pub(crate) use _scproxy::make_module;
+
+#[pymodule]
+mod _scproxy {
+    // straight-forward port of Modules/_scproxy.c
+
+    use crate::builtins::{PyDictRef, PyStr};
+    use crate::pyobject::{IntoPyObject, ItemProtocol, PyResult};
+    use crate::VirtualMachine;
+    use std::ops::Deref;
+    use system_configuration::core_foundation::{
+        array::CFArray,
+        base::TCFType,
+        dictionary::CFDictionary,
+        number::CFNumber,
+        string::{CFString, CFStringRef},
+    };
+    use system_configuration::sys::{
+        dynamic_store_copy_specific::SCDynamicStoreCopyProxies, schema_definitions::*,
+    };
+
+    #[pyfunction]
+    fn _get_proxy_settings(vm: &VirtualMachine) -> PyResult<Option<PyDictRef>> {
+        // Py_BEGIN_ALLOW_THREADS
+        let proxy_dict_raw = unsafe { SCDynamicStoreCopyProxies(std::ptr::null()) };
+        // Py_END_ALLOW_THREADS
+        if proxy_dict_raw.is_null() {
+            return Ok(None);
+        }
+        let proxy_dict =
+            unsafe { CFDictionary::<_, CFNumber>::wrap_under_get_rule(proxy_dict_raw) };
+
+        let result = vm.ctx.new_dict();
+        let v = 0
+            != if let Some(a_num) =
+                unsafe { proxy_dict.find(kSCPropNetProxiesExcludeSimpleHostnames) }
+            {
+                let num: &CFNumber = a_num.deref();
+                num.to_i32().unwrap_or(0)
+            } else {
+                0
+            };
+        result.set_item("exclude_simple", vm.ctx.new_bool(v), vm)?;
+
+        let proxy_dict =
+            unsafe { CFDictionary::<_, CFArray<CFString>>::wrap_under_create_rule(proxy_dict_raw) };
```

Comment:
I think you can just use the `proxy_dict` from before; the high-level [get_proxies](https://docs.rs/system-configuration/0.4.0/system_configuration/dynamic_store/struct.SCDynamicStore.html#method.get_proxies) function returns it as a `CFDictionary<CFString, CFType>`, and I think `find` still works regardless of the V type parameter for CFDictionary